server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -162,22 +163,22 @@ AppenderatorAddResult add(
    * @return future that resolves when all pending data to segments of the identifiers has been persisted, contains
    * commit metadata for this persist
    */
-  ListenableFuture<Object> persist(Collection<SegmentIdentifier> identifiers, Committer committer);
+  ListenableFuture<Object> persist(Collection<SegmentIdentifier> identifiers, @Nullable Committer committer);
```

Comment:
why would persist be called with null committer when it would be a noop? shouldn't persist call throw IAE when committer is null ?

Reply:
Its behavior is different. With the noop committer, Appenderator commits intermediate state which involving writing data on local disk. If the committer is null, Appenderator skips committing intermediate state and writes nothing. Since IndexTask is not restorable, committing intermediate state is unnecessary.

Reply:
IndexTask should be able to persist intermediate data ingested so far to avoid out of disk problem. I guess you were confused because of the wrong javadoc. See https://github.com/druid-io/druid/pull/5297#discussion_r167054485.